### ISSUE TITLE
Error when doing evaluate

### ISSUE THREAD

**Issue by eminarcissus** [jarfo/kchar#2]
Hi,

I'm trying to train the model like this
python train.py --save_every 1 --data_dir data/test --n_chars 10000 --batch
_size 80
Here's the output
creating an LSTM-CNN with  2  layers

---
# Layer (type)                     Output Shape          Param #     Connected to

chars (InputLayer)               (80, 35, 65)          0

---

timedistributed_1 (TimeDistribute(80, 35, 65, 15)      88200       chars[0][0]

---

convolution2d_1 (Convolution2D)  (80, 35, 65, 50)      800         timedistributed_1[0][0]

---

convolution2d_2 (Convolution2D)  (80, 35, 64, 100)     3100        timedistributed_1[0][0]

---

convolution2d_3 (Convolution2D)  (80, 35, 63, 150)     6900        timedistributed_1[0][0]

---

convolution2d_4 (Convolution2D)  (80, 35, 62, 200)     12200       timedistributed_1[0][0]

---

convolution2d_5 (Convolution2D)  (80, 35, 61, 200)     15200       timedistributed_1[0][0]

---

convolution2d_6 (Convolution2D)  (80, 35, 60, 200)     18200       timedistributed_1[0][0]

---

convolution2d_7 (Convolution2D)  (80, 35, 59, 200)     21200       timedistributed_1[0][0]

---

maxpooling2d_1 (MaxPooling2D)    (80, 35, 1, 50)       0           convolution2d_1[0][0]

---

maxpooling2d_2 (MaxPooling2D)    (80, 35, 1, 100)      0           convolution2d_2[0][0]

---

maxpooling2d_3 (MaxPooling2D)    (80, 35, 1, 150)      0           convolution2d_3[0][0]

---

maxpooling2d_4 (MaxPooling2D)    (80, 35, 1, 200)      0           convolution2d_4[0][0]

---

maxpooling2d_5 (MaxPooling2D)    (80, 35, 1, 200)      0           convolution2d_5[0][0]

---

maxpooling2d_6 (MaxPooling2D)    (80, 35, 1, 200)      0           convolution2d_6[0][0]

---

maxpooling2d_7 (MaxPooling2D)    (80, 35, 1, 200)      0           convolution2d_7[0][0]

---

merge_1 (Merge)                  (80, 35, 1, 1100)     0           maxpooling2d_1[0][0]
                                                                   maxpooling2d_2[0][0]
                                                                   maxpooling2d_3[0][0]
                                                                   maxpooling2d_4[0][0]
                                                                   maxpooling2d_5[0][0]
                                                                   maxpooling2d_6[0][0]
                                                                   maxpooling2d_7[0][0]

---

reshape_1 (Reshape)              (80, 35, 1100)        0           merge_1[0][0]

---

timedistributed_2 (TimeDistribute(80, 35, 1100)        2422200     reshape_1[0][0]

---

timedistributed_3 (TimeDistribute(80, 35, 1100)        2422200     timedistributed_2[0][0]

---

lstm_1 (LSTM)                    (80, 35, 650)         4552600     timedistributed_3[0][0]

---

dropout_1 (Dropout)              (80, 35, 650)         0           lstm_1[0][0]

---

lstm_2 (LSTM)                    (80, 35, 650)         3382600     dropout_1[0][0]

---

dropout_2 (Dropout)              (80, 35, 650)         0           lstm_2[0][0]

---
# timedistributed_4 (TimeDistribute(80, 35, 30000)       19530000    dropout_2[0][0]

Total params: 32475400
It outputs a lsm_char_large_epoch20.h5 which I have renamed it to char-large.h5 and evaluate it with
python evaluate.py --model cv/char-large --vocabulary data/test/vocab.npz --text data/test/test.txt --calc

Word vocab size: 30000, Char vocab size: 5880

---
# Layer (type)                     Output Shape          Param #     Connected to

chars (InputLayer)               (1, 1, 65)            0

---

timedistributed_1 (TimeDistribute(1, 1, 65, 15)        88200       chars[0][0]

---

convolution2d_1 (Convolution2D)  (1, 1, 65, 50)        800         timedistributed_1[0][0]

---

convolution2d_2 (Convolution2D)  (1, 1, 64, 100)       3100        timedistributed_1[0][0]

---

convolution2d_3 (Convolution2D)  (1, 1, 63, 150)       6900        timedistributed_1[0][0]

---

convolution2d_4 (Convolution2D)  (1, 1, 62, 200)       12200       timedistributed_1[0][0]

---

convolution2d_5 (Convolution2D)  (1, 1, 61, 200)       15200       timedistributed_1[0][0]

---

convolution2d_6 (Convolution2D)  (1, 1, 60, 200)       18200       timedistributed_1[0][0]

---

convolution2d_7 (Convolution2D)  (1, 1, 59, 200)       21200       timedistributed_1[0][0]

---

maxpooling2d_1 (MaxPooling2D)    (1, 1, 1, 50)         0           convolution2d_1[0][0]

---

maxpooling2d_2 (MaxPooling2D)    (1, 1, 1, 100)        0           convolution2d_2[0][0]

---

maxpooling2d_3 (MaxPooling2D)    (1, 1, 1, 150)        0           convolution2d_3[0][0]

---

maxpooling2d_4 (MaxPooling2D)    (1, 1, 1, 200)        0           convolution2d_4[0][0]

---

maxpooling2d_5 (MaxPooling2D)    (1, 1, 1, 200)        0           convolution2d_5[0][0]

---

maxpooling2d_6 (MaxPooling2D)    (1, 1, 1, 200)        0           convolution2d_6[0][0]

---

maxpooling2d_7 (MaxPooling2D)    (1, 1, 1, 200)        0           convolution2d_7[0][0]

---

merge_1 (Merge)                  (1, 1, 1, 1100)       0           maxpooling2d_1[0][0]
                                                                   maxpooling2d_2[0][0]
                                                                   maxpooling2d_3[0][0]
                                                                   maxpooling2d_4[0][0]
                                                                   maxpooling2d_5[0][0]
                                                                   maxpooling2d_6[0][0]
                                                                   maxpooling2d_7[0][0]

---

reshape_1 (Reshape)              (1, 1, 1100)          0           merge_1[0][0]

---

timedistributed_2 (TimeDistribute(1, 1, 1100)          2422200     reshape_1[0][0]

---

timedistributed_3 (TimeDistribute(1, 1, 1100)          2422200     timedistributed_2[0][0]

---

lstm_1 (LSTM)                    (1, 1, 650)           4552600     timedistributed_3[0][0]

---

dropout_1 (Dropout)              (1, 1, 650)           0           lstm_1[0][0]

---

lstm_2 (LSTM)                    (1, 1, 650)           3382600     dropout_1[0][0]

---

dropout_2 (Dropout)              (1, 1, 650)           0           lstm_2[0][0]

---
# timedistributed_4 (TimeDistribute(1, 1, 30000)         19530000    dropout_2[0][0]

Total params: 32475400

And then it raise out a error like this.
Error when checking model target: expected timedistributed_4 to have shape (1, 1, 30000) but got array with shape (4, 1, 1)
Is there any problem with the model it outputs?

**Comment by jarfo** [comment]
Hi
Some months ago, I temporaly modified the train script as a workaround to a Keras problem with the sparse_categorical_crossentropy loss (but I did not modify the evaluate script).
Since Keras has no problem now with the more efficient sparse_categorical_crossentropy, I have switched back to the sparse_categorical_crossentropy model. (that evaluate.py is also expecting as input model)
You may need to retrain your model with the updated github sourceBest RegardsJose Fonollosa

```
  De: eminarcissus <notifications@github.com>
```

 Para: jarfo/kchar kchar@noreply.github.com 
 Enviado: Sábado 27 de agosto de 2016 7:41
 Asunto: [jarfo/kchar] Error when doing evaluate (#2)

Hi,I'm trying to train the model like this
python train.py --save_every 1 --data_dir data/test --n_chars 10000 --batch
_size 80
Here's the output
creating an LSTM-CNN with 2 layers
Layer (type) Output Shape Param # Connected to
chars (InputLayer) (80, 35, 65) 0timedistributed_1 (TimeDistribute(80, 35, 65, 15) 88200 chars[0][0]convolution2d_1 (Convolution2D) (80, 35, 65, 50) 800 timedistributed_1[0][0]convolution2d_2 (Convolution2D) (80, 35, 64, 100) 3100 timedistributed_1[0][0]convolution2d_3 (Convolution2D) (80, 35, 63, 150) 6900 timedistributed_1[0][0]convolution2d_4 (Convolution2D) (80, 35, 62, 200) 12200 timedistributed_1[0][0]convolution2d_5 (Convolution2D) (80, 35, 61, 200) 15200 timedistributed_1[0][0]convolution2d_6 (Convolution2D) (80, 35, 60, 200) 18200 timedistributed_1[0][0]convolution2d_7 (Convolution2D) (80, 35, 59, 200) 21200 timedistributed_1[0][0]maxpooling2d_1 (MaxPooling2D) (80, 35, 1, 50) 0 convolution2d_1[0][0]maxpooling2d_2 (MaxPooling2D) (80, 35, 1, 100) 0 convolution2d_2[0][0]maxpooling2d_3 (MaxPooling2D) (80, 35, 1, 150) 0 convolution2d_3[0][0]maxpooling2d_4 (MaxPooling2D) (80, 35, 1, 200) 0 convolution2d_4[0][0]maxpooling2d_5 (MaxPooling2D) (80, 35, 1, 200) 0 convolution2d_5[0][0]maxpooling2d_6 (MaxPooling2D) (80, 35, 1, 200) 0 convolution2d_6[0][0]maxpooling2d_7 (MaxPooling2D) (80, 35, 1, 200) 0 convolution2d_7[0][0]merge_1 (Merge) (80, 35, 1, 1100) 0 maxpooling2d_1[0][0]
 maxpooling2d_2[0][0]
 maxpooling2d_3[0][0]
 maxpooling2d_4[0][0]
 maxpooling2d_5[0][0]
 maxpooling2d_6[0][0]
 maxpooling2d_7[0][0]reshape_1 (Reshape) (80, 35, 1100) 0 merge_1[0][0]timedistributed_2 (TimeDistribute(80, 35, 1100) 2422200 reshape_1[0][0]timedistributed_3 (TimeDistribute(80, 35, 1100) 2422200 timedistributed_2[0][0]lstm_1 (LSTM) (80, 35, 650) 4552600 timedistributed_3[0][0]dropout_1 (Dropout) (80, 35, 650) 0 lstm_1[0][0]lstm_2 (LSTM) (80, 35, 650) 3382600 dropout_1[0][0]dropout_2 (Dropout) (80, 35, 650) 0 lstm_2[0][0]
timedistributed_4 (TimeDistribute(80, 35, 30000) 19530000 dropout_2[0][0]
Total params: 32475400
It outputs a lsm_char_large_epoch20.h5 which I have renamed it to char-large.h5 and evaluate it with
python evaluate.py --model cv/char-large --vocabulary data/test/vocab.npz --text data/test/test.txt --calcWord vocab size: 30000, Char vocab size: 5880
Layer (type) Output Shape Param # Connected to
chars (InputLayer) (1, 1, 65) 0timedistributed_1 (TimeDistribute(1, 1, 65, 15) 88200 chars[0][0]convolution2d_1 (Convolution2D) (1, 1, 65, 50) 800 timedistributed_1[0][0]convolution2d_2 (Convolution2D) (1, 1, 64, 100) 3100 timedistributed_1[0][0]convolution2d_3 (Convolution2D) (1, 1, 63, 150) 6900 timedistributed_1[0][0]convolution2d_4 (Convolution2D) (1, 1, 62, 200) 12200 timedistributed_1[0][0]convolution2d_5 (Convolution2D) (1, 1, 61, 200) 15200 timedistributed_1[0][0]convolution2d_6 (Convolution2D) (1, 1, 60, 200) 18200 timedistributed_1[0][0]convolution2d_7 (Convolution2D) (1, 1, 59, 200) 21200 timedistributed_1[0][0]maxpooling2d_1 (MaxPooling2D) (1, 1, 1, 50) 0 convolution2d_1[0][0]maxpooling2d_2 (MaxPooling2D) (1, 1, 1, 100) 0 convolution2d_2[0][0]maxpooling2d_3 (MaxPooling2D) (1, 1, 1, 150) 0 convolution2d_3[0][0]maxpooling2d_4 (MaxPooling2D) (1, 1, 1, 200) 0 convolution2d_4[0][0]maxpooling2d_5 (MaxPooling2D) (1, 1, 1, 200) 0 convolution2d_5[0][0]maxpooling2d_6 (MaxPooling2D) (1, 1, 1, 200) 0 convolution2d_6[0][0]maxpooling2d_7 (MaxPooling2D) (1, 1, 1, 200) 0 convolution2d_7[0][0]merge_1 (Merge) (1, 1, 1, 1100) 0 maxpooling2d_1[0][0]
 maxpooling2d_2[0][0]
 maxpooling2d_3[0][0]
 maxpooling2d_4[0][0]
 maxpooling2d_5[0][0]
 maxpooling2d_6[0][0]
 maxpooling2d_7[0][0]reshape_1 (Reshape) (1, 1, 1100) 0 merge_1[0][0]timedistributed_2 (TimeDistribute(1, 1, 1100) 2422200 reshape_1[0][0]timedistributed_3 (TimeDistribute(1, 1, 1100) 2422200 timedistributed_2[0][0]lstm_1 (LSTM) (1, 1, 650) 4552600 timedistributed_3[0][0]dropout_1 (Dropout) (1, 1, 650) 0 lstm_1[0][0]lstm_2 (LSTM) (1, 1, 650) 3382600 dropout_1[0][0]dropout_2 (Dropout) (1, 1, 650) 0 lstm_2[0][0]
timedistributed_4 (TimeDistribute(1, 1, 30000) 19530000 dropout_2[0][0]
Total params: 32475400And then it raise out a error like this.
Error when checking model target: expected timedistributed_4 to have shape (1, 1, 30000) but got array with shape (4, 1, 1)
Is there any problem with the model it outputs? —
